common/reviews/api/rush-lib.api.md

```diff
@@ -663,14 +745,46 @@ export class VersionPolicyConfiguration {
     get versionPolicies(): Map<string, VersionPolicy>;
 }
 
-// @beta
+// @public
 export enum VersionPolicyDefinitionName {
     // (undocumented)
     'individualVersion' = 1,
     // (undocumented)
     'lockStepVersion' = 0
 }
 
+// @beta
+export class WebClient {
```

Comment:
The `WebClient` API seems very specific to Rush's internals.  Also, it is coupled with the `node-fetch` typings (which are now somewhat controversial since the maintainer made a strange decision to drop support for CommonJS).

We should either make this `@internal` or refactor it to be more general purpose.  I'm marking it as `@beta` only to unblock the release.

Reply:
What about moving it to node-core-library and putting together our own typings for anything we're reexporting from `node-fetch`? We can then just stick the user agent in the constructor.

Reply:
**(Out of scope for this PR)**

That's not a bad idea.  The Rush Stack projects do relatively few HTTP client requests, so we actually haven't put a lot of thought into the right way.  There are several options:

- Node.js exposes native APIs for this, but they're different from the browser runtime, and (last I checked) they are not even promisified
- The standard is `node-fetch` (which ponyfills the browser standard `fetch()`), but as I mentioned that project looks a bit wacky right now.
- The other super popular option is [Axios](https://www.npmjs.com/package/axios) which is a uniform solution for browser and Node.js, and adds [several nice features](https://blog.logrocket.com/axios-or-fetch-api/) (cancellation, timeouts, download progress bars, etc) that go beyond `fetch()`.  The main tradeoff is that it's another dependency 18k minified, 63k for the debug bundle.
- Lastly, if you are writing Node.js services then you'll be using a framework like [Fastify](https://www.fastify.io/) or [Hapi](https://hapi.dev/) which have their own integrated client APIs

`node-core-library` aims to provide a centralized toolkit that enables developers to perform common operations, using best practices, without having to think through these choices themselves.  So for example, if we decided that `fetch()` is the best practice, I agree that it would make sense for `node-core-library` to provide a readymade wrapper for it because it is a tiny API and standardized.  Whereas if we choose one of the other 3 options above, I think we'd simply write docs somewhere saying "we recommend to use that thing."

Reply:
@iclanton and I chatted about this yesterday and decided to temporarily copy+paste `WebClient` to avoid adding it to the `rush-lib` API surface, since we're sure wherever it ends up, it won't be there.